### PR TITLE
feature: support runtime init script

### PR DESCRIPTION
**Pull request by VoVAllen**
support `runtime.init`

### CODE REVIEW

**Comment by VoVAllen** [comment]
@gaocegege @kemingy PTAL

File: pkg/lang/frontend/starlark/runtime/const.go [hunk 1]
```diff
@@ -15,9 +15,10 @@
...
-	ruleExpose  = "runtime.expose"
-	ruleDaemon  = "runtime.daemon"
-	ruleEnviron = "runtime.environ"
-	ruleMount   = "runtime.mount"
+	ruleCommand    = "runtime.command"
+	ruleExpose     = "runtime.expose"
+	ruleDaemon     = "runtime.daemon"
+	ruleEnviron    = "runtime.environ"
+	ruleMount      = "runtime.mount"
+	ruleInitScript = "runtime.init_script"
```
Comment (VoVAllen) [inline pkg/lang/frontend/starlark/runtime/const.go:23]: Is it better to name as `runtime.init`?

Comment (kemingy) [reply pkg/lang/frontend/starlark/runtime/const.go:23]: SGTM

File: pkg/lang/ir/supervisor.go [hunk 2]
```diff
@@ -28,11 +28,14 @@ import (
...
-stdout = "/var/logs/%[1]s_stdout.log"
-stderr = "/var/logs/%[1]s_stderr.log"
+command = """
+%[2]s
+"""
+stdout = "/var/log/horust/%[1]s_stdout.log"
+stderr = "/var/log/horust/%[1]s_stderr.log"
 user = "${USER}"
 working-directory = "${%[3]s}"
+%[4]s
```
Comment (kemingy) [inline pkg/lang/ir/supervisor.go:38]: I wonder if this is too flexible. Currently, we only need the `start-after`.

Maybe we can switch to a TOML tool to generate it if it becomes more complex in the future.

File: pkg/lang/ir/supervisor.go [hunk 3]
```diff
@@ -60,22 +74,29 @@ func (g Graph) compileEntrypoint(root llb.State) llb.State {
...
+		for i, command := range g.RuntimeInitScript {
+			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("init_%d", i), fmt.Sprintf("/bin/bash -c 'set -euo pipefail\n%s'", strings.Join(command, "\n")), nil)
+			deps = append(deps, fmt.Sprintf("init_%d", i))
+		}
+	}
 
 	if g.RuntimeDaemon != nil {
 		for i, command := range g.RuntimeDaemon {
-			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("daemon_%d", i), fmt.Sprintf("%s &\n", strings.Join(command, " ")))
+			entrypoint = g.addNewProcess(entrypoint, fmt.Sprintf("daemon_%d", i), fmt.Sprintf("%s &\n", strings.Join(command, " ")), deps)
```
Comment (kemingy) [inline pkg/lang/ir/supervisor.go:88]: I think we should emphasize their dependency relationship in the document later.

Comment (gaocegege) [reply pkg/lang/ir/supervisor.go:88]: Let's open an issue in envd-docs to keep track.

Comment (kemingy) [reply pkg/lang/ir/supervisor.go:88]: https://github.com/tensorchord/envd-docs/issues/170

**Comment by gaocegege** [comment]
LGTM

**Comment by gaocegege** [comment]
Sorry, I closed it in accident.

File: pkg/lang/ir/supervisor.go (hunk 2 above)
Comment (VoVAllen) [reply pkg/lang/ir/supervisor.go:38]: We can refactor this once we need more controls here.

File: examples/dpgen2/build.envd [hunk 4]
```diff
@@ -0,0 +1,40 @@
+def install_kubectl_kind():
```
Comment (gaocegege) [inline examples/dpgen2/build.envd:1]: Should we keep it in envd examples or envdlib?

Comment (VoVAllen) [reply examples/dpgen2/build.envd:1]: It's not general enough now. `sudo docker network connect kind dpgen2` means only project named as dpgen2 will work

Comment (gaocegege) [reply examples/dpgen2/build.envd:1]: Could we make like this:

```
def install_kubectl_kind(environment_name):
```

Comment (VoVAllen) [reply examples/dpgen2/build.envd]: Changed to `def install_kubectl_kind(env_name)`. But I still think `kind` is not a general requirement. Thus I don't think we don't need to put it into envdlib

**Review by gaocegege — APPROVED** [review]

**Review by kemingy — APPROVED** [review]
/lgtm

**Pull request merged**